### DISCUSSION TITLE
Float or double? Vdac

### DISCUSSION THREAD

**Discussion by Laiaborrell** [TheAlbertDev/MASB#60] [Q&A] [answered]
Hi!
The `void MCP4725_SetOutputVoltage(MCP4725_Handle_T self, float outputVoltage)` function takes as input a float variable representing the Vdac that we want to set. However, when we receive the `CA_configuration` the `eDC` variable to which we want to set the Vcell is a double. It is true that we then have to convert the Vcell to the Vdac with the formula given in the readme, but we don't know if the Vdac variable:
```Vdac=1.65-(Vcell/2);```
must be defined as float or double.
I hope the question makes sense. Thank you in advance.
Laia

**Comment by Albert-Alvarez** [comment, marked as answer]
As you say, `eDc` is a `double` and `Vdac` is a `float`. Simply, operate with `eDc` the relevant formula and, when saving the result in the variable `Vdac`, it will be cast directly to `float`. You can also put it explicitly:

```c
float Vdac = (float)(1.65-(eDc/2.0));
```
